src/core/mesh/qgsmeshdataprovider.h

```diff
@@ -102,6 +102,20 @@ struct CORE_EXPORT QgsMesh
     */
   void clear();
 
+  /**
+    * Returns the vertex position in the face, -1 if the face do not exist or no such vertex in the face
+    *
+    * \since QGIS 3.22
+    */
+  int vertexPosition( int vertexIndex, int faceIndex );
```

Comment:
ufff.... I think the name of the function is misleading. "position" suggest it is lat/long or something else. 'index' would be probably more appropriate, but it is not vertex index, but some local index within face...

I am thinking if it is really necessary to add these methods to the public interface? Maybe just define them inline static in the cpp unit where you need them?